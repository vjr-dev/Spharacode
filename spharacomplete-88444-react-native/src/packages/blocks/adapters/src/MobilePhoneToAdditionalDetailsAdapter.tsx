import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class MobilePhoneToAdditionalDetailsAdapter {
  send: (message: Message) => void;

  constructor() {
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.NavigationMobilePhoneAdditionalDetailsMessage)
    ]);
  }

  convert = (from: Message): Message => {
    const to = new Message(getName(MessageEnum.NavigationMessage));

    to.addData(
      getName(MessageEnum.NavigationTargetMessage),
      'AdditionalDetailForm'
    );

    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );

    const raiseMessage: Message = new Message(
      getName(MessageEnum.NavigationPayLoadMessage)
    );
    raiseMessage.addData(
      getName(MessageEnum.AuthTokenDataMessage),
      from.getData(getName(MessageEnum.AuthTokenDataMessage))
    );

    to.addData(getName(MessageEnum.NavigationRaiseMessage), raiseMessage);

    return to;
  };

  receive(from: string, message: Message): void {
    this.send(this.convert(message));
  }
}
