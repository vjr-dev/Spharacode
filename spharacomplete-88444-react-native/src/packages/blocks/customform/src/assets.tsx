export const imgAdd = require("../assets/ic_add.png");
export const imgCancel = require("../assets/ic_cancel.png");
export const imgGstin = require("../assets/ic_gstin.png");
export const imgMapIcon = require("../assets/ic_map_icon.png");
export const imgSuccessModal = require("../assets/success_modal_image.png");
export const backIcon = require("../assets/backIcon.png");