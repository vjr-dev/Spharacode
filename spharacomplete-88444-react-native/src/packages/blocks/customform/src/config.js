Object.defineProperty(exports, "__esModule", {
  value: true,
});

// Customizable Area Start
exports.validationApiContentType = "application/json";
exports.validationApiMethodType = "GET";
exports.exampleAPiEndPoint = "EXAMPLE_URL";
exports.exampleAPiMethod = "POST";
exports.sellerDetailsApiContentType = "application/json";
exports.sellersAPIEndPoint = "custom_form/seller_accounts";

exports.sellerDetailsAPIMethodPUT = "PUT";
exports.sellerDetailsAPIMethodPOST = "POST";
exports.sellerDetailsAPIMethodPATCH = "PATCH";
exports.getSellersAPIMethod = "GET";
exports.textInputPlaceHolder = "Enter Text";
exports.labelTitleText = "customform";
exports.labelBodyText = "customform Body";
exports.errorMessageShopName = "Shop name is mandatory";
exports.errorMessageGstin = "GSTIN is mandatory";

exports.btnExampleTitle = "CLICK ME";

exports.congrats = "Congratulations!";
exports.formSubmitted = "Your form is submitted";
exports.sellerDetails = "Seller Details";
// Customizable Area End
