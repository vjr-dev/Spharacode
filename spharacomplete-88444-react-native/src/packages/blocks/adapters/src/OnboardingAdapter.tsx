import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class OnboardingAdapter {
  send: (message: Message) => void;

  constructor() {
    // const uuidv4 = require('uuid/v4');
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.OnboardingGuideControllerDoneOnboarding),
      getName(MessageEnum.OnboardingGuideControllerSkipOnboarding)
    ]);
  }

  convert = (from: Message): Message => {
    runEngine.debugLog("OnboardingAdapter", from.id)
    const to = new Message(getName(MessageEnum.NavigationMessage));
    to.addData(getName(MessageEnum.NavigationTargetMessage), 'Home');
    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );
    return to;
  };

  receive(from: string, message: Message): void {
    this.send(this.convert(message));
  }
}
