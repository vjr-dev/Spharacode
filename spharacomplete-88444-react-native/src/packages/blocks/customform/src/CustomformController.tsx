import { IBlock } from "../../../framework/src/IBlock";
import { Message } from "../../../framework/src/Message";
import { BlockComponent } from "../../../framework/src/BlockComponent";
import MessageEnum, {
  getName,
} from "../../../framework/src/Messages/MessageEnum";
import { runEngine } from "../../../framework/src/RunEngine";

// Customizable Area Start
export const Colors = {
  inputLabel: "rgba(31, 31, 34, 0.6)",
  inputTextColor: "rgb(31, 31, 34)",
  borderGrey: "rgba(28, 28, 30, 0.3)",
  borderYellow: "rgb(205, 149, 12)",
  white: "#FFFFFF",
  modalBg: "rgb(243, 243, 243)",
};

const options = {
  title: "Select Image",
  storageOptions: {
    skipBackup: true,
    path: "images",
  },
};
// Customizable Area End

export const configJSON = require("./config");

export interface Props {
  navigation: any;
  id: string;
  // Customizable Area Start
  // Customizable Area End
}

interface S {
  // Customizable Area Start
  getSellerDetailsMessageId: any;
  postSellerDetailsMessageId: any;
  sellerID: any;
  shopName: string;
  address: string;
  gstin: string;
  selectedServices: number[];
  showSuccessModal: boolean;
  token: string;
  lat: any;
  long: any;
  // Customizable Area End
}

interface SS {
  id: any;
}

export default class CustomformController extends BlockComponent<Props, S, SS> {
  constructor(props: Props) {
    super(props);
    this.receive = this.receive.bind(this);

    // Customizable Area Start
    this.subScribedMessages = [
      getName(MessageEnum.RestAPIRequestMessage),
      getName(MessageEnum.RestAPIResponceMessage),
      getName(MessageEnum.NavigationPayLoadMessage),
      getName(MessageEnum.SessionResponseMessage),
    ];
    this.state = {
      getSellerDetailsMessageId: "",
      postSellerDetailsMessageId: "",
      sellerID: "",
      shopName: "",
      address: "",
      gstin: "",
      selectedServices: [],
      token: "",
      showSuccessModal: false,
      lat: 0,
      long: 0,
    };
    // Customizable Area End
    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
  }

  async receive(from: string, message: Message) {
    // Customizable Area Start
    runEngine.debugLog("Message Recived", message);

    if (getName(MessageEnum.SessionResponseMessage) === message.id) {
      let token = message.getData(getName(MessageEnum.SessionResponseToken));
      this.setState({ token: token });
      this.getSellerDetail(token)
    }

    if (getName(MessageEnum.RestAPIResponceMessage) === message.id) {
      const apiRequestCallId = message.getData(
        getName(MessageEnum.RestAPIResponceDataMessage)
      );

      var responseJson = message.getData(
        getName(MessageEnum.RestAPIResponceSuccessMessage)
      );

      var errorReponse = message.getData(
        getName(MessageEnum.RestAPIResponceErrorMessage)
      );

      console.log("API REQUEST CALL ID: ", apiRequestCallId);

      if (apiRequestCallId && responseJson) {
        if (apiRequestCallId === this.state.postSellerDetailsMessageId) {
          this.showModal();
          console.log("Received from post API: ", responseJson);
        }

        if (apiRequestCallId === this.state.getSellerDetailsMessageId) {          
          console.log("Received from get API: ", responseJson);
          if (responseJson.data && responseJson.data.attributes) {
            console.log(
              "responseJson.data.attributes ",
              responseJson.data.attributes
            );
            let isWholesalerSelected =
              responseJson.data.attributes.wholesaler != undefined
                ? responseJson.data.attributes.wholesaler
                : false;
            let isRetailerSelected =
              responseJson.data.attributes.retailer != undefined
                ? responseJson.data.attributes.retailer
                : false;
            let isManufacturerSelected =
              responseJson.data.attributes.manufacturer != undefined
                ? responseJson.data.attributes.manufacturer
                : false;
            let isHallmarking_centerSelected =
              responseJson.data.attributes.hallmarking_center != undefined
                ? responseJson.data.attributes.hallmarking_center
                : false;

            let selectedServices = [];

            if (isWholesalerSelected) {
              selectedServices.push(1);
            }
            if (isManufacturerSelected) {
              selectedServices.push(2);
            }
            if (isHallmarking_centerSelected) {
              selectedServices.push(3);
            }
            if (isRetailerSelected) {
              selectedServices.push(4);
            }

            this.setState({
              sellerID: responseJson.data.id,
              shopName: responseJson.data.attributes.firm_name
                ? responseJson.data.attributes.firm_name
                : "",
              address: responseJson.data.attributes.location
                ? responseJson.data.attributes.location
                : "",
              gstin: responseJson.data.attributes.gstin_number
                ? responseJson.data.attributes.gstin_number
                : "",
              selectedServices: selectedServices,
            });
          }
        }
      }
    } 
    // Customizable Area End
  }

  // Customizable Area Start
  async componentDidMount() {
    super.componentDidMount();
    this.getToken();
    if (this.isPlatformWeb() === false) {
      this.props.navigation.addListener("willFocus", () => {
        this.getToken();
      });
    }
  }
  
  getToken = () => {
    const msg: Message = new Message(
      getName(MessageEnum.SessionRequestMessage)
    );
    this.send(msg);
  };


  labelShopName: string = "Name of Shop";
  labelAddress: string = "Address";
  labelGstin: string = "GSTIN";
  labelAbout: string = "About Us";
  labelServiceProvided: string = "Services Provided";
  labelDealsIn: string = "Deals in";
  labelShopPhotos: string = "Shop Photos(Min 1 - Max 8)";
  labelVisitingCard: string = "Upload Visiting Card";
  btnLabel: string = "Submit";
  btnContinueLabel: string = "Continue";

  services = [
    { label: "Wholesaler", value: 1 },
    { label: "Manufacturer", value: 2 },
    { label: "Hallmarker", value: 3 },
    { label: "Retailer", value: 4 },
  ];

  getSellerDetail = async (token:string) => {
   
    const header = {
      "Content-Type": configJSON.sellerDetailsApiContentType,
      token: token,
    };

    const requestMessage = new Message(
      getName(MessageEnum.RestAPIRequestMessage)
    );

    this.setState({
      getSellerDetailsMessageId: requestMessage.messageId,
    });

    requestMessage.addData(
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      configJSON.sellersAPIEndPoint + "/1"
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      JSON.stringify(header)
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestMethodMessage),
      configJSON.getSellersAPIMethod
    );

    runEngine.sendMessage(requestMessage.id, requestMessage);
  };

  submitSellerDetails = async () => {

    let token = this.state.token;

    if (!token || token.length === 0 ) {
      this.showAlert("Error", "Invaild Token. Plese log in.")
      return;
    }
    if (
      this.state.shopName.trim() == ""
    ) {
      this.showAlert("Error", configJSON.errorMessageShopName);
      return;
    }

    const header = {
      "Content-Type": configJSON.sellerDetailsApiContentType,
    };

    let isWholesalerSelected = false;
    let isRetailerSelected = false;
    let isManufacturerSelected = false;
    let isHallmarking_centerSelected = false;

    this.state.selectedServices.forEach((value) => {
      switch (value) {
        case 1:
          isWholesalerSelected = true;
          break;
        case 2:
          isManufacturerSelected = true;
          break;
        case 3:
          isHallmarking_centerSelected = true;
          break;
        case 4:
          isRetailerSelected = true;
          break;
      }
    });

    const httpBody = {
      token: token,
      seller_account: {
        firm_name: this.state.shopName,
        location: this.state.address,
        gstin_number: this.state.gstin,
        wholesaler: isWholesalerSelected,
        retailer: isRetailerSelected,
        manufacturer: isManufacturerSelected,
        hallmarking_center: isHallmarking_centerSelected,
        lat: this.state.lat,
        long: this.state.long,
      },
    };

    const requestMessage = new Message(
      getName(MessageEnum.RestAPIRequestMessage)
    );

    this.setState({
      postSellerDetailsMessageId: requestMessage.messageId,
    });

    
    requestMessage.addData(
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      this.state.sellerID.length > 1 ? configJSON.sellersAPIEndPoint + "/" + this.state.sellerID : configJSON.sellersAPIEndPoint
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      JSON.stringify(header)
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestBodyMessage),
      JSON.stringify(httpBody)
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestMethodMessage),
      this.state.sellerID.length > 1 ? configJSON.sellerDetailsAPIMethodPUT : configJSON.sellerDetailsAPIMethodPOST
    );

    runEngine.sendMessage(requestMessage.id, requestMessage);
  };

  onChangeInput = (name: string, value: any) => {
    // @ts-ignore
    this.setState({ [name]: value });
  };

  txtInputProps = (key: string) => ({
    // @ts-ignore
    value: this.state[key],
    onChangeText: (text: string) => {
      this.onChangeInput(key, text);
    },
  });

  inputLabelProps = {
    fontSize: 14,
    color: Colors.inputLabel,
    lineHeight: 16,
  };

  onServiceSelected = (serviceId: number) => {
    if (!this.state.selectedServices.includes(serviceId)) {
      this.setState({
        selectedServices: this.state.selectedServices.concat(serviceId),
      });
    }
  };

  onServiceUnSelected = (serviceId: number) => {
    this.setState({
      selectedServices: this.state.selectedServices.filter(
        (selectedService) => selectedService !== serviceId
      ),
    });
  };

  showModal = () => {
    this.setState({ showSuccessModal: true });
  };

  hideModal = () => {
    this.setState({ showSuccessModal: false }, () => {
    });
  };
  // Customizable Area End
}
