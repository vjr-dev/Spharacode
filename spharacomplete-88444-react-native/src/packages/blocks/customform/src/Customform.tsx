import React from "react";
// Customizable Area Start
import {
  StyleSheet,
  Image,
  TouchableOpacity,
  View,
  TextInput,
  ScrollView,
  TouchableWithoutFeedback,
  Platform,
  Modal,
  Dimensions,
  Text
} from "react-native";

import {
  imgCancel,
  imgSuccessModal
} from "./assets";

import { Colors } from "./CustomformController";

import GenericLabel from "../../../components/src/GenericLabel";

const screenWidth = Math.round(Dimensions.get("window").width);
// Customizable Area End

import CustomformController, {
  Props,
  configJSON,
} from "./CustomformController";

export default class Customform extends CustomformController {
  constructor(props: Props) {
    super(props);
    // Customizable Area Start
    // Customizable Area End
  }

  // Customizable Area Start
  renderModal = () => {
    return (
      <Modal visible={this.state.showSuccessModal} transparent>
        <View style={styles.modalWapper}>
          <View style={styles.modalContent}>
            <GenericLabel fontSize={24} lineHeight={32} align="center">
              {configJSON.congrats}
            </GenericLabel>
            <GenericLabel fontSize={24} lineHeight={32} align="center">
              {configJSON.formSubmitted}
            </GenericLabel>
            <View style={styles.modalImageWrapper}>
              <Image source={imgSuccessModal} style={styles.successImage} />
            </View>
            <TouchableOpacity
              style={styles.continueBtn}
              onPress={this.hideModal}
            >
              <GenericLabel
                fontSize={16}
                color={Colors.white}
                onPress={this.hideModal}
              >
                {this.btnContinueLabel}
              </GenericLabel>
            </TouchableOpacity>
          </View>
        </View>
      </Modal>
    );
  };
  // Customizable Area End

  render() {
    return (
      //Merge Engine DefaultContainer
      <ScrollView keyboardShouldPersistTaps="always" style={styles.container}>
        <TouchableWithoutFeedback
          testID="btnHideKeyboard"
          onPress={() => {
            this.hideKeyboard();
          }}
        >
          {/* Customizable Area Start */}
          {/* Merge Engine UI Engine Code */}
          <View>
            <View style={styles.contentContainer}>
              <View style={styles.formInputWrapper}>
                <GenericLabel {...this.inputLabelProps}>
                  {this.labelShopName}
                </GenericLabel>
                <View style={styles.inputWrapper}>
                  <TextInput
                    testID="shopName"
                    style={styles.textInput}
                    {...this.txtInputProps("shopName")}
                  />
                </View>
              </View>
              <View style={styles.formInputWrapper}>
                <GenericLabel {...this.inputLabelProps}>
                  {this.labelServiceProvided}
                </GenericLabel>
                <View style={styles.servicesWrapper}>
                  {this.services.map((service, index) => (
                    <View
                      style={styles.serviceItemWrapper}
                      key={`SERVICES_${index}`}
                    >
                      <SelectionWrapper
                        isSelected={this.state.selectedServices.includes(
                          service.value
                        )}
                        option={service}
                        onPress={() => this.onServiceSelected(service.value)}
                        onClearPress={() =>
                          this.onServiceUnSelected(service.value)
                        }
                      />
                    </View>
                  ))}
                </View>
              </View>
              <TouchableOpacity
                testID="submitButton"
                style={styles.submitBtn}
                onPress={() => this.submitSellerDetails()}
              >
                <Text>{this.btnLabel}</Text>
              </TouchableOpacity>
              {this.renderModal()}
            </View>
          </View>
          {/* Merge Engine UI Engine Code */}
          {/* Customizable Area End */}
        </TouchableWithoutFeedback>
      </ScrollView>
      //Merge Engine End DefaultContainer
    );
  }
}

// Customizable Area Start
// @ts-ignore
const SelectionWrapper = ({ option, isSelected, onPress, onClearPress }) => {
  return (
    <TouchableOpacity onPress={onPress}>
      <View
        style={[
          styles.selectionWrapper,
          { borderColor: isSelected ? Colors.borderYellow : Colors.borderGrey },
        ]}
      >
        <GenericLabel
          color={isSelected ? Colors.borderYellow : "rgb(46, 58, 89)"}
          onPress={onPress}
        >
          {option.label}
        </GenericLabel>
        {isSelected && (
          <TouchableWithoutFeedback>
            <TouchableOpacity onPress={onClearPress}>
              <Image source={imgCancel} style={styles.clearIcon} />
            </TouchableOpacity>
          </TouchableWithoutFeedback>
        )}
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#ffffffff",
  },
  contentContainer: {
    flex: 1,
    paddingHorizontal: 16,
    paddingVertical: 20,
    marginLeft: "auto",
    marginRight: "auto",
    width: Platform.OS === "web" ? "75%" : "100%",
    maxWidth: 650,
    backgroundColor: "#ffffffff",
  },
  content: {
    width: "100%",
  },
  formInputWrapper: {
    width: "100%",
    marginBottom: 20,
  },
  inputWrapper: {
    borderBottomColor: Colors.borderGrey,
    borderBottomWidth: 1,
    flexDirection: "row",
    alignItems: "center",
  },
  textInput: {
    height: 48,
    width: "100%",
    flex: 1,
    fontSize: 16,
    color: Colors.inputTextColor,
    fontWeight: "600",
  },
  selectionWrapper: {
    borderWidth: 1,
    borderRadius: 5,
    paddingVertical: 12,
    paddingHorizontal: 12,
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
  },
  clearIcon: {
    width: 16,
    height: 16,
  },
  servicesWrapper: {
    marginTop: 10,
    flexDirection: "row",
    flexWrap: "wrap",
    width: "100%",
    justifyContent: "space-between",
  },
  serviceItemWrapper: {
    width: "48%",
    marginBottom: 16,
  },
  submitBtn: {
    backgroundColor: Colors.borderYellow,
    alignItems: "center",
    justifyContent: "center",
    paddingVertical: 16,
    width: "100%",
    marginBottom: 20,
    borderRadius: 4,
  },
  shopImagePlusWrapper: {
    height: 80,
    width: 80,
    borderWidth: 1,
    borderColor: Colors.borderYellow,
    alignItems: "center",
    justifyContent: "center",
    borderRadius: 3,
    marginRight: 12,
    marginBottom: 12,
  },
  shopImagePlusIcon: {
    height: 20,
    width: 20,
  },
  selectedImageWrapper: {
    height: 80,
    width: 80,
    borderRadius: 3,
    marginRight: 12,
    marginBottom: 12,
  },
  selectedImage: {
    height: 80,
    width: 80,
    borderRadius: 3,
  },
  modalWapper: {
    backgroundColor: "rgba(255, 255, 255, 0.6)",
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    paddingHorizontal: 18,
  },
  modalContent: {
    backgroundColor: Colors.modalBg,
    width: "100%",
    borderRadius: 10,
    paddingHorizontal: 16,
    paddingVertical: 32,
  },
  successImage: {
    width: 264,
    height: 158,
  },
  modalImageWrapper: {
    alignItems: "center",
    justifyContent: "center",
    width: "100%",
  },
  continueBtn: {
    backgroundColor: Colors.borderYellow,
    alignItems: "center",
    justifyContent: "center",
    paddingVertical: 16,
    width: "100%",
    marginTop: 20,
    borderRadius: 4,
  },
});
// Customizable Area End
