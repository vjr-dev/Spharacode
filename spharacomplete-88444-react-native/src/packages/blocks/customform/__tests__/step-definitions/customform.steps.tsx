import { defineFeature, loadFeature } from "jest-cucumber";
import { shallow, ShallowWrapper } from "enzyme";

import * as helpers from "../../../../framework/src/Helpers";
import { runEngine } from "../../../../framework/src/RunEngine";
import { Message } from "../../../../framework/src/Message";

import MessageEnum, {
  getName,
} from "../../../../framework/src/Messages/MessageEnum";
import React from "react";
import Customform from "../../src/Customform";

const screenProps = {
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    push: jest.fn(),
    pop: jest.fn(),
    replace: jest.fn(),
  },
  id: "Customform",
};

const feature = loadFeature("./__tests__/features/customform-scenario.feature");

defineFeature(feature, (test) => {
  beforeEach(() => {
    jest.resetModules();
    jest.doMock("react-native", () => ({ Platform: { OS: "android" } }));
    jest.spyOn(helpers, "getOS").mockImplementation(() => "");
  });

  test("User navigates to customform", ({ given, when, then }) => {
    let customformBlock: ShallowWrapper;
    let instance: Customform;

    given("I am a User loading customform", () => {
      customformBlock = shallow(<Customform {...screenProps} />);
    });

    when("I navigate to the customform", () => {
      instance = customformBlock.instance() as Customform;
    });

    then("customform will load with out errors", () => {
      expect(customformBlock).toBeTruthy();
      expect(customformBlock).toMatchSnapshot();

    ///


      const tokenMsg: Message = new Message(getName(MessageEnum.SessionResponseMessage));
      tokenMsg.addData(getName(MessageEnum.SessionResponseToken), "TOKEN");
      runEngine.sendMessage("Unit Test", tokenMsg);

      const msgValidationAPI = new Message(getName(MessageEnum.RestAPIResponceMessage))
      msgValidationAPI.addData(getName(MessageEnum.RestAPIResponceDataMessage), msgValidationAPI.messageId);
      msgValidationAPI.addData(getName(MessageEnum.RestAPIResponceSuccessMessage), 
        {"data":{"id":"8","type":"seller_account","attributes":{"firm_name":"Shop NAME  122","full_phone_number":null,"location":"us","country_code":null,"phone_number":null,"gstin_number":"-1","wholesaler":true,"retailer":true,"manufacturer":false,"hallmarking_center":false,"buy_gold":null,"buy_silver":null,"sell_gold":null,"sell_silver":null,"activated":false}}}
      );

      instance.setState({
        getSellerDetailsMessageId: msgValidationAPI.messageId,
      });

      runEngine.sendMessage("Unit Test", msgValidationAPI)
    
    });

    then("I can enter firm name with out errors", () => {
      let textInputComponent = customformBlock.findWhere(
        (node) => node.prop("testID") === "shopName"
      );
      textInputComponent.simulate("changeText", "Gold Gully");
      expect(customformBlock).toBeTruthy();
    });

    then("I can select the button with with out errors", () => {
      let buttonComponent = customformBlock.findWhere(
        (node) => node.prop("testID") === "submitButton"
      );
      buttonComponent.simulate("press");
      expect(instance.state.shopName).toEqual("Gold Gully");
    });

    then("I can leave the screen with out errors", () => {

      let buttonComponent = customformBlock.findWhere(
        (node) => node.prop("testID") === "btnHideKeyboard"
      );
      buttonComponent.simulate("press");
      
      instance.componentWillUnmount();
      expect(customformBlock).toBeTruthy();
    });
  });
});
