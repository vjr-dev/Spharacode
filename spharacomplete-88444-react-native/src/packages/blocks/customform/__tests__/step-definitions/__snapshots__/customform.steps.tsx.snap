// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customform User navigates to customform 1`] = `
<ScrollView
  keyboardShouldPersistTaps="always"
  style={
    Object {
      "backgroundColor": "#ffffffff",
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    onPress={[Function]}
    testID="btnHideKeyboard"
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#ffffffff",
            "flex": 1,
            "marginLeft": "auto",
            "marginRight": "auto",
            "maxWidth": 650,
            "paddingHorizontal": 16,
            "paddingVertical": 20,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "marginBottom": 20,
              "width": "100%",
            }
          }
        >
          <GenericLabel
            align="left"
            color="rgba(31, 31, 34, 0.6)"
            fontSize={14}
            isUpperCase={false}
            lineHeight={16}
            mb={0}
            ml={0}
            mr={0}
            mt={0}
            pb={0}
            pl={0}
            pr={0}
            pt={0}
            underline={false}
          >
            Name of Shop
          </GenericLabel>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "rgba(28, 28, 30, 0.3)",
                "borderBottomWidth": 1,
                "flexDirection": "row",
              }
            }
          >
            <Component
              allowFontScaling={true}
              onChangeText={[Function]}
              rejectResponderTermination={true}
              style={
                Object {
                  "color": "rgb(31, 31, 34)",
                  "flex": 1,
                  "fontSize": 16,
                  "fontWeight": "600",
                  "height": 48,
                  "width": "100%",
                }
              }
              testID="shopName"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
        <View
          style={
            Object {
              "marginBottom": 20,
              "width": "100%",
            }
          }
        >
          <GenericLabel
            align="left"
            color="rgba(31, 31, 34, 0.6)"
            fontSize={14}
            isUpperCase={false}
            lineHeight={16}
            mb={0}
            ml={0}
            mr={0}
            mt={0}
            pb={0}
            pl={0}
            pr={0}
            pt={0}
            underline={false}
          >
            Services Provided
          </GenericLabel>
          <View
            style={
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "marginTop": 10,
                "width": "100%",
              }
            }
          >
            <View
              key="SERVICES_0"
              style={
                Object {
                  "marginBottom": 16,
                  "width": "48%",
                }
              }
            >
              <SelectionWrapper
                isSelected={false}
                onClearPress={[Function]}
                onPress={[Function]}
                option={
                  Object {
                    "label": "Wholesaler",
                    "value": 1,
                  }
                }
              />
            </View>
            <View
              key="SERVICES_1"
              style={
                Object {
                  "marginBottom": 16,
                  "width": "48%",
                }
              }
            >
              <SelectionWrapper
                isSelected={false}
                onClearPress={[Function]}
                onPress={[Function]}
                option={
                  Object {
                    "label": "Manufacturer",
                    "value": 2,
                  }
                }
              />
            </View>
            <View
              key="SERVICES_2"
              style={
                Object {
                  "marginBottom": 16,
                  "width": "48%",
                }
              }
            >
              <SelectionWrapper
                isSelected={false}
                onClearPress={[Function]}
                onPress={[Function]}
                option={
                  Object {
                    "label": "Hallmarker",
                    "value": 3,
                  }
                }
              />
            </View>
            <View
              key="SERVICES_3"
              style={
                Object {
                  "marginBottom": 16,
                  "width": "48%",
                }
              }
            >
              <SelectionWrapper
                isSelected={false}
                onClearPress={[Function]}
                onPress={[Function]}
                option={
                  Object {
                    "label": "Retailer",
                    "value": 4,
                  }
                }
              />
            </View>
          </View>
        </View>
        <ForwardRef
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(205, 149, 12)",
              "borderRadius": 4,
              "justifyContent": "center",
              "marginBottom": 20,
              "paddingVertical": 16,
              "width": "100%",
            }
          }
          testID="submitButton"
        >
          <Text>
            Submit
          </Text>
        </ForwardRef>
        <Component
          hardwareAccelerated={false}
          transparent={true}
          visible={false}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(255, 255, 255, 0.6)",
                "flex": 1,
                "justifyContent": "center",
                "paddingHorizontal": 18,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "rgb(243, 243, 243)",
                  "borderRadius": 10,
                  "paddingHorizontal": 16,
                  "paddingVertical": 32,
                  "width": "100%",
                }
              }
            >
              <GenericLabel
                align="center"
                color="rgb(46, 58, 89)"
                fontSize={24}
                isUpperCase={false}
                lineHeight={32}
                mb={0}
                ml={0}
                mr={0}
                mt={0}
                pb={0}
                pl={0}
                pr={0}
                pt={0}
                underline={false}
              >
                Congratulations!
              </GenericLabel>
              <GenericLabel
                align="center"
                color="rgb(46, 58, 89)"
                fontSize={24}
                isUpperCase={false}
                lineHeight={32}
                mb={0}
                ml={0}
                mr={0}
                mt={0}
                pb={0}
                pl={0}
                pr={0}
                pt={0}
                underline={false}
              >
                Your form is submitted
              </GenericLabel>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../packages/blocks/customform/assets/success_modal_image.png",
                    }
                  }
                  style={
                    Object {
                      "height": 158,
                      "width": 264,
                    }
                  }
                />
              </View>
              <ForwardRef
                onPress={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "rgb(205, 149, 12)",
                    "borderRadius": 4,
                    "justifyContent": "center",
                    "marginTop": 20,
                    "paddingVertical": 16,
                    "width": "100%",
                  }
                }
              >
                <GenericLabel
                  align="left"
                  color="#FFFFFF"
                  fontSize={16}
                  isUpperCase={false}
                  mb={0}
                  ml={0}
                  mr={0}
                  mt={0}
                  onPress={[Function]}
                  pb={0}
                  pl={0}
                  pr={0}
                  pt={0}
                  underline={false}
                >
                  Continue
                </GenericLabel>
              </ForwardRef>
            </View>
          </View>
        </Component>
      </View>
    </View>
  </TouchableWithoutFeedback>
</ScrollView>
`;
